src/main/java/no/hvl/dat250/h2020/group5/service/PollService.java

```diff
@@ -47,14 +47,18 @@ public PollResponse createPoll(Poll poll, Long userId) {
 
   public boolean deletePoll(Long pollId, Long userId) {
     Optional<Poll> poll = pollRepository.findById(pollId);
-    if (poll.isEmpty()) {
+    if (poll.isEmpty() || !isOwnerOrAdmin(poll.get(), userId)) {
       return false;
     }
-    if (isOwnerOrAdmin(poll.get(), userId)) {
-      pollRepository.delete(poll.get());
-      return true;
+
+    for (Vote vote : voteRepository.findByPoll(poll.get())) {
+      vote.setVoterAndAddThisVoteToVoter(null);
+      vote.setPollAndAddThisVoteToPoll(null);
```

Comment:
Burde ikke vi bruke `poll.deleteVote(vote)` og `vote.setVoter(null)` og `vote.setPoll(null)` for å fjerne avhengigheten her? For nå blir vel vote bare lagt til i poll igjen? Har prøvd noe lignende i 
#88 men da failet IT testene 😬